contrib/tracking.go

```diff
@@ -0,0 +1,256 @@
+package contrib
+
+/*
+#include "tracking.h"
+*/
+import "C"
+import (
+	"image"
+
+	"gocv.io/x/gocv"
+)
+
+// Tracker - the base interface for object tracking
+//
+// s.a: https://docs.opencv.org/master/d0/d0a/classcv_1_1Tracker.html
+type Tracker interface {
+	// Init initializes the tracker with a known bounding box that surrounded the target
+	// note: this can only be called once. if you loose the object, you have to Close() the instance,
+	// create a new one, and call Init on it again
+	//
+	// https://docs.opencv.org/master/d0/d0a/classcv_1_1Tracker.html#a4d285747589b1bdd16d2e4f00c3255dc
+	Init(image gocv.Mat, boundingBox image.Rectangle) bool
+
+	// Update the tracker, return new bounding box and a bool, if it lost it.
+	//
+	// https://docs.opencv.org/master/d0/d0a/classcv_1_1Tracker.html#a549159bd0553e6a8de356f3866df1f18
+	Update(image gocv.Mat) (image.Rectangle, bool)
+}
+
+//
+// available implementations in gocv are:
+//
+// TrackerMil;
+// TrackerBoosting;
+// TrackerMedianFlow;
+// TrackerTld;
+// TrackerKcf;
+// TrackerMosse;
+// TrackerCsrt;
+//
+
+// (private) implementation helpers for init and update (saves some boilerplate code)
+func tracker_Init(trk C.Tracker, img gocv.Mat, boundingBox image.Rectangle) bool {
+	cBox := C.struct_Rect{
+		x:      C.int(boundingBox.Min.X),
+		y:      C.int(boundingBox.Min.Y),
+		width:  C.int(boundingBox.Size().X),
+		height: C.int(boundingBox.Size().Y),
+	}
+
+	ret := C.Tracker_Init(trk, C.Mat(img.Ptr()), cBox)
+	return bool(ret)
+}
+
+func tracker_Update(trk C.Tracker, img gocv.Mat) (image.Rectangle, bool) {
+	cBox := C.struct_Rect{}
+
+	ret := C.Tracker_Update(trk, C.Mat(img.Ptr()), &cBox)
+
+	rect := image.Rect(int(cBox.x), int(cBox.y), int(cBox.x+cBox.width), int(cBox.y+cBox.height))
+	return rect, bool(ret)
+}
+
+//
+// Mil
+// The MIL algorithm trains a classifier in an online manner to separate the object from the background.
```

Comment:
It should starts with:

> TrackerMil implements the MIL algorithm, which trains a classifier [...]

It is important because here is what I currently get on godoc:

![screen shot 2018-03-12 at 12 13 27](https://user-images.githubusercontent.com/217628/37280732-ef088fa2-25ee-11e8-9db5-84e4b9786a37.png)

I care a lot about documentation because this is the only thing this project offers me (as a user) to use it.

Reply:
This comment applies to all trackers then.

Reply:
> what I currently get on godoc

oh, haven't tried at all, but sure, it should make sense in the godoc context (vs humans reading src code)

Reply:
why stylesheet? I use `godoc -http 127.0.0.1:6060` only.